eclipse-che/che-server
# adapt kubernetes routing to nginx ingress controller

### What does this PR do?
nginx ingress controller (not community one) started requiring `nginx.org/websocket-services` annotations.
So, this PR add an ability to use `<<service-name>>` placeholder in ingress annotations, and make helm chart using it.

### Screenshot/screencast of this PR
![Screenshot_20210716_115825](https://user-images.githubusercontent.com/5887312/125921936-15a9405f-064d-400d-94eb-2bde859dc94a.png)


### What issues does this PR fix or reference?
See users complaining about it in https://github.com/eclipse/che/issues/19434
https://github.com/eclipse/che/issues/19796

But I assume there may be other places where it should be done, like Che operator, gateway

### How to test this PR?
1. Deploy on minikube with helm chart ./home/sleshche/projects/che-server/deploy/kubernetes/helm/che/charts /home/sleshche/projects/che-server/deploy/kubernetes/helm/che/tmpcharts.
2. Start a workspace and make sure workspace ingress has the needed annotation.


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
